spring-cloud-gcp-vision/src/main/java/org/springframework/cloud/gcp/vision/CloudVisionTemplate.java

```diff
@@ -61,10 +72,15 @@ public CloudVisionTemplate(ImageAnnotatorClient imageAnnotatorClient) {
 	 * @param featureTypes the types of image analysis to perform on the image.
 	 * @return the results of image analyses
 	 */
-	public AnnotateImageResponse analyzeImage(Resource imageResource, Feature.Type... featureTypes)
-			throws IOException {
+	public AnnotateImageResponse analyzeImage(Resource imageResource, Feature.Type... featureTypes) {
+		ByteString imgBytes;
+		try {
+			imgBytes = ByteString.readFrom(imageResource.getInputStream());
+		}
+		catch (IOException e) {
+			throw new CloudVisionException(e);
```

Comment:
Should also provide a message like "Failed to read image bytes from provided resource: <path>".